pkg/cmd/delete/clusterset/exec.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Contributors to the Open Cluster Management project
+package clusterset
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	clusterclientset "open-cluster-management.io/api/client/cluster/clientset/versioned"
+	"open-cluster-management.io/clusteradm/pkg/helpers"
+)
+
+func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
+
+	o.Clustersets = args
+
+	return nil
+}
+
+func (o *Options) validate() (err error) {
+	if len(o.Clustersets) == 0 {
+		return fmt.Errorf("the name of the clusterset must be specified")
+	}
+	if len(o.Clustersets) > 1 {
+		return fmt.Errorf("only one clusterset can be deleted")
+	}
+	return nil
+}
+
+func (o *Options) run() (err error) {
+	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	clusterClient, err := clusterclientset.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	clusterSetName := o.Clustersets[0]
+
+	return o.runWithClient(clusterClient, o.ClusteradmFlags.DryRun, clusterSetName)
+}
+
+// check unband first
+
+func (o *Options) runWithClient(clusterClient clusterclientset.Interface,
+	dryRun bool,
+	clusterset string) error {
+
+	// check existing
+	_, err := clusterClient.ClusterV1beta1().ManagedClusterSets().Get(context.TODO(), clusterset, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			fmt.Fprintf(o.Streams.Out, "Clusterset %s is already deleted\n", clusterset)
+			return nil
+		}
+		return err
+	}
+
+	// check binding
+	list, err := clusterClient.ClusterV1beta1().ManagedClusterSetBindings(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
```

Comment:
you might want a fieldSelector here to check if the binding is to this specified clusterset

Reply:
@qiujian16  done. 
https://github.com/open-cluster-management-io/clusteradm/pull/154/files#diff-f7baf7874e7a100baa3867cec75f246850885661199413546017eddbe00a6891R66

Reply:
```
╰─$ clusteradm get clustersets                         
NAME      BOUND NAMESPACES   STATUS
default                      1 ManagedClusters selected
set1      default            No ManagedCluster selected
set2      default            No ManagedCluster selected

╰─$ clusteradm delete clusterset set1
Clusterset set1 still bind to a namespace! Please unbind before deleted.

╰─$ clusteradm clusterset unbind set1 --namespace default
Clusterset default is unbind from Clusterset set1

╰─$ clusteradm delete clusterset set1                    
Clusterset set1 is deleted

╰─$ clusteradm get clustersets       
NAME      BOUND NAMESPACES   STATUS
default                      1 ManagedClusters selected
set2      default            No ManagedCluster selected
```